graph/nodes_edges.go

```diff
@@ -0,0 +1,203 @@
+	for it.Next() {
+		n = append(n, it.Node())
+	}
+	return n
+}
+
+// Edges is an Edge iterator.
+type Edges interface {
+	Iterator
+
+	// Edge returns the current Edge from the iterator.
+	Edge() Edge
+}
+
+type EdgeSlicer interface {
+	// EdgeSlice returns the set of edges remaining
+	// to be iterated by an Edges iterator.
+	// The holder of the iterator may arbitrarily
+	// change elements in the returned slice, but
+	// those changes may be reflected to other
+	// iterators.
+	EdgeSlice() []Edge
+}
+
+// NodesOf returns it.Len() nodes from it. If it is an EdgeSlicer, the EdgeSlice method is used
+// to obtain the edges. It is safe to pass a nil Edges to EdgesOf.
+func EdgesOf(it Edges) []Edge {
+	if it == nil {
+		return nil
+	}
+	switch it := it.(type) {
+	case EdgeSlicer:
+		return it.EdgeSlice()
+	}
+	n := make([]Edge, 0, it.Len())
+	for it.Next() {
+		n = append(n, it.Edge())
+	}
+	return n
+}
+
+// WeightedEdges is an Edge iterator.
+type WeightedEdges interface {
+	Iterator
+
+	// Edge returns the current Edge from the iterator.
+	WeightedEdge() WeightedEdge
+}
+
+type WeightedEdgeSlicer interface {
+	// EdgeSlice returns the set of edges remaining
+	// to be iterated by an Edges iterator.
+	// The holder of the iterator may arbitrarily
+	// change elements in the returned slice, but
+	// those changes may be reflected to other
+	// iterators.
+	WeightedEdgeSlice() []WeightedEdge
+}
+
+// WeightedEdgesOf returns it.Len() weighted edge from it. If it is an WeightedEdgeSlicer, the
+// WeightedEdgeSlice method is used to obtain the edges. It is safe to pass a nil Edges to
+// WeightedEdgesOf.
+func WeightedEdgesOf(it WeightedEdges) []WeightedEdge {
+	if it == nil {
+		return nil
+	}
+	switch it := it.(type) {
+	case WeightedEdgeSlicer:
+		return it.WeightedEdgeSlice()
+	}
+	n := make([]WeightedEdge, 0, it.Len())
+	for it.Next() {
+		n = append(n, it.WeightedEdge())
+	}
+	return n
+}
+
+// Lines is an Line iterator.
+type Lines interface {
+	Iterator
+
+	// Line returns the current Line from the iterator.
+	Line() Line
+}
+
+type LineSlicer interface {
+	// LineSlice returns the set of lines remaining
+	// to be iterated by an Lines iterator.
+	// The holder of the iterator may arbitrarily
+	// change elements in the returned slice, but
+	// those changes may be reflected to other
+	// iterators.
+	LineSlice() []Line
+}
+
+// NodesOf returns it.Len() nodes from it. If it is an LineSlicer, the LineSlice method is used
+// to obtain the lines. It is safe to pass a nil Lines to LinesOf.
+func LinesOf(it Lines) []Line {
+	if it == nil {
+		return nil
+	}
+	switch it := it.(type) {
+	case LineSlicer:
+		return it.LineSlice()
+	}
+	n := make([]Line, 0, it.Len())
+	for it.Next() {
+		n = append(n, it.Line())
+	}
+	return n
+}
+
+// WeightedLines is an Line iterator.
+type WeightedLines interface {
+	Iterator
+
+	// Line returns the current Line from the iterator.
+	WeightedLine() WeightedLine
+}
+
+type WeightedLineSlicer interface {
```

Comment:
Missing doc comment for WeightedLineSlicer, etc (seems to be all FooSlicer's).

Reply:
Indeed. <kbd>CTRL-C</kbd><kbd>CTRL-V</kbd>.

Fixed.